### PR TITLE
Vrp auth flow persistence flow

### PR DESCRIPTION
**Pull request by kalpanakanagasabai**
## VRP Persistence flow

> PR contains the implementations defined to handle consent persistence based on the consent types.

**Issue link:** #38  

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** OB Accelerator

------

### Development Checklist

1. [x] Build complete solution with pull request in place.
2. [ ] Ran checkstyle plugin with pull request in place.
3. [ ] Ran Findbugs plugin with pull request in place.
4. [ ] Ran FindSecurityBugs plugin and verified report.
5. [x] Formatted code according to WSO2 code style.
6. [x] Have you verified the PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [x] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [x] Written unit tests.
2. [ ] Verified tests in multiple database environments (if applicable).
3. [ ] Tested with BI enabled  (if applicable).

### CODE REVIEW

File: open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/persistence/flow/ConsentPersistStepTests.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
+/**
```
Comment (Ashi1993) [inline open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/persistence/flow/ConsentPersistStepTests.java:1]: Shall we change the license year

File: open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/utils/ConsentAuthorizeTestConstants.java [hunk 2]
```diff
@@ -319,4 +327,92 @@ public class ConsentAuthorizeTestConstants {
...
+                    "       \"cofAccount\":\"\"," +
+                    "       \"approval\":\"true\"," +
+                    "       \"accountIds\":[\"\"]," +
+                    "       \"isReauthorization\":\"\"," +
+                    "       \"type\":\"payments\"," +
+                    "       \"paymentAccount\":\"30080012343456\"," +
+                    "       \"MultiAuthType\":\"Any\"," +
+                    "       \"MultiAuthExpiry\":\"1626755005019\"" +
+                    "   }";
+
```
Comment (Ashi1993) [inline open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/utils/ConsentAuthorizeTestConstants.java:408]: Remove these extra lines

Comment (kalpanakanagasabai) [reply open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/utils/ConsentAuthorizeTestConstants.java]: Fixed in [909ecfe]

File: open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/persistence/flow/ConsentPersistStepTests.java (hunk 1 above)
Comment (kalpanakanagasabai) [reply open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/persistence/flow/ConsentPersistStepTests.java:1]: Fixed in [909ecfe]

File: open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/impl/DefaultConsentPersistStep.java [hunk 3]
```diff
@@ -66,16 +70,57 @@ public void execute(ConsentPersistData consentPersistData) throws ConsentExcepti
...
-            consentPersistenceHandler.consentPersist(consentPersistData, consentResource);
 
         } catch (ConsentManagementException e) {
             throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
                     "Exception occured while persisting consent");
         }
     }
+
+    /**
+     * Abstract method defined to handle consent persistence based on the consent type.
```
Comment (Ashi1993) [inline open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/impl/DefaultConsentPersistStep.java:83]: This is not an abstract method

Comment (kalpanakanagasabai) [reply open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/impl/DefaultConsentPersistStep.java]: Fixed in [e6c1916]

File: open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/persistence/flow/ConsentPersistStepTests.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+        JSONObject payload = (JSONObject) parser
+                .parse(ConsentAuthorizeTestConstants.COF_PERSIST_PAYLOAD_WITH_NON_STRING_COF_ACC);
+        doReturn(payload).when(consentPersistDataMock).getPayload();
+
+        consentPersistStep.execute(consentPersistDataMock);
+    }
+
+    @Test(priority = 11, expectedExceptions = ConsentException.class)
+    public void testCOFPersistThrowingExceptionWhenConsentBinding() throws Exception {
```
Comment (hasithakn) [inline open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/persistence/flow/ConsentPersistStepTests.java:210]: Shall we add a test with happy path too?, (Test without expecting an exception)

Comment (kalpanakanagasabai) [reply open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/persistence/flow/ConsentPersistStepTests.java:219]: Fixed in [f1f5354]

**Review by Ashi1993 — APPROVED** [review]

**Review by hasithakn — APPROVED** [review]

**Pull request merged**